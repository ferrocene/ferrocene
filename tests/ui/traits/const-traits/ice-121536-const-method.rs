#![feature(const_trait_impl, effects)] //~ WARN the feature `effects` is incomplete

pub struct Vec3;

#[const_trait]
pub trait Add {
    fn add(self) -> Vec3;
}

impl Add for Vec3 {
    const fn add(self) -> Vec3 {
        //~^ ERROR functions in trait impls cannot be declared const
        self
    }
}

fn main() {}
