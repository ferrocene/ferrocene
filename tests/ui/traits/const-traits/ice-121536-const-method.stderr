error[E0379]: functions in trait impls cannot be declared const
  --> $DIR/ice-121536-const-method.rs:11:5
   |
LL |     const fn add(self) -> Vec3 {
   |     ^^^^^ functions in trait impls cannot be const
   |
help: remove the `const` ...
   |
LL -     const fn add(self) -> Vec3 {
LL +     fn add(self) -> Vec3 {
   |
help: ... and declare the impl to be const instead
   |
LL | impl const Add for Vec3 {
   |      +++++

warning: the feature `effects` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/ice-121536-const-method.rs:1:30
   |
LL | #![feature(const_trait_impl, effects)]
   |                              ^^^^^^^
   |
   = note: see issue #102090 <https://github.com/rust-lang/rust/issues/102090> for more information
   = note: `#[warn(incomplete_features)]` on by default

error: using `#![feature(effects)]` without enabling next trait solver globally
   |
   = note: the next trait solver must be enabled globally for the effects feature to work correctly
   = help: use `-Znext-solver` to enable

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0379`.
